data_structures/src/chain/priority.rs

```diff
@@ -0,0 +1,1240 @@
+╟──────────┬───────────────┬───────────────────────────────║
+║     Tier │ Time-to-block │ Priority                      ║
+╟──────────┼───────────────┼───────────────────────────────║
+║   Stinky │ {:>13} │ {:<28}  ║
+║      Low │ {:>13} │ {:<28}  ║
+║   Medium │ {:>13} │ {:<28}  ║
+║     High │ {:>13} │ {:<28}  ║
+║  Opulent │ {:>13} │ {:<28}  ║
+╠══════════════════════════════════════════════════════════╣
+║ Value transfer transactions                              ║
+╟──────────┬───────────────┬───────────────────────────────║
+║     Tier │ Time-to-block │ Priority                      ║
+╟──────────┼───────────────┼───────────────────────────────║
+║   Stinky │ {:>13} │ {:<28}  ║
+║      Low │ {:>13} │ {:<28}  ║
+║   Medium │ {:>13} │ {:<28}  ║
+║     High │ {:>13} │ {:<28}  ║
+║  Opulent │ {:>13} │ {:<28}  ║
+╚══════════════════════════════════════════════════════════╝"#,
+            // Believe it or not, these `to_string` are needed for proper formatting, hence the
+            // clippy allow directive above.
+            self.drt_stinky.time_to_block.to_string(),
+            self.drt_stinky.priority.to_string(),
+            self.drt_low.time_to_block.to_string(),
+            self.drt_low.priority.to_string(),
+            self.drt_medium.time_to_block.to_string(),
+            self.drt_medium.priority.to_string(),
+            self.drt_high.time_to_block.to_string(),
+            self.drt_high.priority.to_string(),
+            self.drt_opulent.time_to_block.to_string(),
+            self.drt_opulent.priority.to_string(),
+            self.vtt_stinky.time_to_block.to_string(),
+            self.vtt_stinky.priority.to_string(),
+            self.vtt_low.time_to_block.to_string(),
+            self.vtt_low.priority.to_string(),
+            self.vtt_medium.time_to_block.to_string(),
+            self.vtt_medium.priority.to_string(),
+            self.vtt_high.time_to_block.to_string(),
+            self.vtt_high.priority.to_string(),
+            self.vtt_opulent.time_to_block.to_string(),
+            self.vtt_opulent.priority.to_string(),
+        )
+    }
+}
+
+/// A estimate for priority and time-to-block.
+///
+/// Time-to-block states what is the expected time (in epochs) that it would take for a transaction
+/// with this priority to be included into a block.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PriorityEstimate {
+    pub priority: Priority,
+    pub time_to_block: TimeToBlock,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct TimeToBlock(u64);
+
+impl TimeToBlock {
+    pub fn from_secs(secs: u64) -> Self {
+        Self(secs)
+    }
+}
+
+impl fmt::Display for TimeToBlock {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&seconds_to_human_string(self.0))
+    }
+}
+
+#[inline]
+fn option_update(option: &mut Option<Priority>, candidate: Option<Priority>) {
```

Comment:
I don't understand the purpose of this function, I can see what it does but not why does it do it that way. A one line comment would help.

Reply:
Check the documentation and example I added. I hope that helps!

Reply:
Thanks, much better! I was confused because the `match (&candidate, &option) {` uses the arguments in reverse order `option: &mut Option<Priority>, candidate: Option<Priority>`, so I was actually understanding it wrong.